### PR TITLE
feat: [CDS-40392]:create-new-env-in-envgroup

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
Added option to create new environment inside environment group modal.

#### Screenshots
Screenshot 2023-03-01 at 10 45 59
Screenshot 2023-03-01 at 10 46 27
Screenshot 2023-03-01 at 10 46 39

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/EnvironmentGroups/CreateEnvironmentGroupModal.tsx [hunk 1]
```diff
@@ -294,14 +295,21 @@ export default function CreateEnvironmentGroupModal({
                       onChange={setSearchTerm}
                       throttle={200}
                     />
-                    <Layout.Vertical height={300} className={css.modalEnvironmentList}>
+                    <Layout.Vertical height={300} className={css.modalEnvironmentList} spacing={'small'}>
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/EnvironmentGroups/CreateEnvironmentGroupModal.tsx:298]: nit: spacing may not be required

**Review by chetanvekariya-harness — DISMISSED** [review]

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger cypress-cv

File: src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupDetails/EditEnvironmentGroupModal.tsx [hunk 2]
```diff
@@ -67,11 +68,24 @@ export default function EditEnvironmentGroupModal({
...
+              newEnvOpen={newEnvOpen}
+              handleNewEnvModal={status => setNewEnvOpen(status)}
+            />
+          </Container>
+          <Button
+            onClick={() => setNewEnvOpen(true)}
+            variation={ButtonVariation.LINK}
+            width={200}
+            icon="plus"
+            padding={{ top: 'small' }}
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupDetails/EditEnvironmentGroupModal.tsx:85]: nit: margin-bottom on Container is better than padding-top on button. padding would give that much extra space for user to click and may not be nice

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

File: src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupDetails/EditEnvironmentGroupModal.tsx (hunk 2 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupDetails/EditEnvironmentGroupModal.tsx:85]: +1

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by litvinovstas** [comment]
retrigger jest

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**